webapp/converter.xqm

```diff
@@ -3,27 +3,33 @@ module namespace e = 'some.namespace';
 declare
   %rest:path("/v1tov2")
   %rest:POST("{$xml}")
-  %output:method("xml")
-  %output:indent("no")
-  %output:omit-xml-declaration("no")
-  %output:doctype-public("-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD with MathML3 v1.2 20190208//EN")
-  %output:doctype-system("JATS-archivearticle1-mathml3.dtd")
-function e:v1tov2($xml as document-node())
+function e:v1tov2($xml as item())
 {
-  let $xsl := doc('./xsl/v1-to-v2.xsl') 
-  return xslt:transform($xml,$xsl,map{'indent':'no'})
+  let $options := map{'indent':'no',
+                    'omit-xml-declaration':'no',
+                    'doctype-public':'-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD with MathML3 v1.2 20190208//EN',
+                    'doctype-system':'JATS-archivearticle1-mathml3.dtd'}
+  let $xsl := doc('./xsl/v1-to-v2.xsl')
+  let $new-xml := e:strip-preamble($xml)
+  return xslt:transform-text($new-xml,$xsl,$options) 
 };
 
 declare
   %rest:path("/v2tov1")
   %rest:POST("{$xml}")
-  %output:method("xml")
-  %output:indent("no")
-  %output:omit-xml-declaration("no")
-  %output:doctype-public("-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD v1.1 20151215//EN")
-  %output:doctype-system("JATS-archivearticle1.dtd")
-function e:v2tov1($xml as document-node())
+function e:v2tov1($xml as item())
 {
-  let $xsl := doc('./xsl/v2-to-v1.xsl') 
-  return xslt:transform($xml,$xsl,map{'indent':'no'})
+  let $options := map{'indent':'no',
+                  'omit-xml-declaration':'no',
+                  'doctype-public':'-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD v1.1 20151215//EN',
+                  'doctype-system':'JATS-archivearticle1.dtd'}
+  let $xsl := doc('./xsl/v2-to-v1.xsl')
+  let $new-xml := e:strip-preamble($xml)
+  return xslt:transform-text($new-xml,$xsl,$options)
+};
+
+declare function e:strip-preamble($xml){
+  if (contains($xml,'<?covid-19-tdm ?>'))
+    then ('<?covid-19-tdm'||substring-after($xml,'<?covid-19-tdm'))
+  else ('<article'||substring-after($xml,'<article'))
```

Comment:
This seems to leave passed xml as just `<article` instead of contacting `substring-after($xml,'<article')`